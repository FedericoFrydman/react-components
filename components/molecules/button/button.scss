@import '../../assets/styles/brand.scss';
@import '../../assets/styles/colors.scss';

/* Button States */

@mixin default($bgColor, $borderColor, $textColor) {
  background-color: $bgColor;
  box-shadow: 0 0 0 1.5px $borderColor;
  color: $textColor;
}

@mixin active($bgColor, $borderColor, $textColor) {
  background-color: $bgColor;
  box-shadow: 0 0 0 1.5px $borderColor;
  color: $textColor;
}

@mixin disabled($bgColor, $borderColor, $textColor) {
  &:disabled,
  &:disabled:hover,
  body.using-key &:disabled:focus,
  &:disabled:active {
    background-color: $bgColor;
    box-shadow: 0 0 0 1.5px $borderColor;
    color: $textColor;
    cursor: not-allowed;
    opacity: 0.5;
  }
}

@mixin hover($bgColor, $borderColor, $haloColor, $textColor) {
  background-color: $bgColor;
  box-shadow: 0 0 0 1.5px $haloColor;
  color: $textColor;
  transition: box-shadow ease-in 0.15s;
}

@mixin focus($bgColor, $borderColor, $haloColor, $textColor) {
  background-color: $bgColor;
  box-shadow: 0 0 0 1.5px $borderColor, 0 0 0 5px $haloColor;
  color: $textColor;
  transition: box-shadow ease-out 0.15s;
}

/* Button Styling */

.lsux-button {
  align-items: center;
  border: 4px solid rgba(255, 255, 255, 0);
  box-shadow: 0 0 0 0 rgba(255, 255, 255, 0);
  -moz-box-sizing: border-box;
  -webkit-box-sizing: border-box;
  box-sizing: border-box;
  cursor: pointer;
  display: flex;
  font-weight: 600;
  height: auto;
  justify-content: center;
  outline: none;
  position: relative;
  transition: all ease-in-out 0.1s;
  width: auto;

  > .lsux-icon {
    + span {
      margin-left: 5px;
    }
  }

  > .lsux-text {
    align-items: center;
    display: flex;
    letter-spacing: -0.011em;
    line-height: 22px;
    margin: 0;
    padding: 0 0.25rem;
  }

  > span {
    + .lsux-icon {
      margin-left: 10px;
    }
  }

  &--icon-only {
    border: none;
    &-no-pad {
      padding: 0px;
    }
  }

  &--stretch {
    width: 100%;
  }

  /* Button Shapes */

  &--rectangular {
    border-radius: 9px;
  }
  &--round {
    border-radius: 50px;
  }

  /* Button Padding */

  &--primary,
  &--secondary {
    padding: 7px 6px;
  }
  &--tertiary {
    padding: 2px 6px;
  }

  /* Button Variants */

  &--primary {
    @include default(
      var(--btn-primary-default-background),
      var(--btn-primary-default-border),
      var(--btn-primary-default-text)
    );

    @include disabled(
      var(--btn-primary-disabled-text),
      var(--btn-primary-disabled-background),
      var(--btn-primary-disabled-border)
    );

    &:hover {
      @include hover(
        var(--btn-primary-hover-background),
        var(--btn-primary-hover-border),
        var(--btn-primary-default-border),
        var(--btn-primary-hover-text)
      );
    }

    body.using-key &:focus,
    &:focus {
      @include focus(
        var(--btn-primary-focus-background),
        var(--btn-primary-focus-border),
        var(--btn-primary-focus-halo),
        var(--btn-primary-focus-text)
      );
    }

    &:active {
      @include active(
        var(--btn-primary-active-background),
        var(--btn-primary-active-border),
        var(--btn-primary-active-text)
      );
    }

    &.lsux-button--destructive {
      @include default($R100, $R100, $N00);

      @include disabled($N00, $R100, $R100);

      &:hover {
        @include hover(lighten($R100, 5%), $R100, transparent, $N00);
      }

      body.using-key &:focus,
      &:focus {
        @include focus(lighten($R100, 5%), lighten($R100, 35%), lighten($R100, 35%), $N00);
      }

      &:active {
        @include active(lighten($R100, 8%), $R100, $N00);
      }
    }
  }

  &--secondary {
    @include default(
      var(--btn-secondary-default-background),
      var(--btn-secondary-default-border),
      var(--btn-secondary-default-text)
    );

    @include disabled(
      var(--btn-secondary-disabled-text),
      var(--btn-secondary-disabled-background),
      var(--btn-secondary-disabled-border)
    );

    &:hover {
      @include hover(
        var(--btn-secondary-hover-background),
        var(--btn-secondary-hover-border),
        var(--btn-secondary-default-border),
        var(--btn-secondary-hover-text)
      );
    }

    body.using-key &:focus,
    &:focus {
      @include focus(
        var(--btn-secondary-focus-background),
        var(--btn-secondary-focus-border),
        var(--btn-secondary-focus-halo),
        var(--btn-secondary-focus-text)
      );
    }

    &:active {
      @include active(
        var(--btn-secondary-active-background),
        var(--btn-secondary-active-border),
        var(--btn-secondary-active-text)
      );
    }

    > .flex-row {
      align-items: center;
    }

    &.lsux-button--destructive {
      @include default(transparent, $R100, $R100);

      @include disabled($R100, rgba(0, 0, 0, 0), $R100);

      &:hover {
        box-shadow: 0 0 0 1.5px lighten($R100, 5%);
        color: lighten($R100, 5%);
      }

      body.using-key &:focus,
      &:focus {
        @include focus($N00, $R100, lighten($R100, 35%), $R100);
      }

      &:active {
        @include active($N00, $R100, lighten($R100, 7%));
      }
    }
  }

  &--tertiary {
    @include default(
      var(--btn-tertiary-default-background),
      var(--btn-tertiary-default-border),
      var(--btn-tertiary-default-text)
    );

    @include disabled(
      var(--btn-tertiary-disabled-background),
      var(--btn-tertiary-disabled-border),
      var(--btn-tertiary-disabled-text)
    );

    &:hover {
      @include hover(
        var(--btn-tertiary-hover-background),
        var(--btn-tertiary-hover-border),
        var(--btn-tertiary-default-border),
        var(--btn-tertiary-hover-text)
      );
    }

    body.using-key &:focus,
    &:focus {
      @include focus(
        var(--btn-tertiary-focus-background),
        var(--btn-tertiary-focus-border),
        var(--btn-tertiary-focus-halo),
        var(--btn-tertiary-focus-text)
      );
    }

    &:active {
      @include active(
        var(--btn-tertiary-active-background),
        var(--btn-tertiary-active-border),
        var(--btn-tertiary-active-text)
      );
    }

    &.lsux-button--destructive {
      @include default(transparent, $N00, $R100);

      @include disabled($R100, rgba(0, 0, 0, 0), rgba(0, 0, 0, 0));

      &:hover {
        color: lighten($R100, 5%);
      }

      body.using-key &:focus,
      &:focus {
        @include focus($N00, $R100, lighten($R100, 35%), lighten($R100, 5%));
      }

      &:active {
        box-shadow: 0 0 0 1.5px lighten($R100, 8%);
        color: lighten($R100, 7%);
      }
    }
  }
}
