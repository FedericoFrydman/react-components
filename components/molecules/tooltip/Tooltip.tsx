import Tippy from '@tippyjs/react';
import { forwardRef, Ref } from 'react';
import cn from 'classnames';

import { Text, TextSize } from '../../atoms/text/Text';

import 'tippy.js/animations/perspective.css';
import 'tippy.js/animations/scale.css';
import 'tippy.js/animations/shift-away.css';
import 'tippy.js/animations/shift-toward.css';
import 'tippy.js/dist/tippy.css';
import 'tippy.js/themes/light.css';
import 'tippy.js/themes/light-border.css';
import 'tippy.js/themes/material.css';
import 'tippy.js/themes/translucent.css';

export type TooltipPlacement = 'top' | 'right' | 'bottom' | 'left';
export type TooltipAnimation = 'shift-away' | 'shift-toward' | 'scale' | 'perspective' | '';
export type TooltipTheme = 'light' | 'light-border' | 'material' | 'translucent';

export interface TooltipProps extends Omit<React.HTMLProps<HTMLDivElement>, 'css'> {
  /**
   * Text of the tooltip
   */
  text: string;
  /**
   * Hovered element
   */
  children: JSX.Element;
  /**
   * The ref to the HTML DOM element
   */
  ref?: Ref<HTMLDivElement>;
  /**
   * Text size of the button
   */
  textSize?: TextSize;
  /**
   * Custom classes for the component.
   */
  classNames?: cn.Argument;
  /**
   * Placement of the tooltip
   */
  placement?: TooltipPlacement;
  /**
   * Shows arrow when true
   */
  arrow?: boolean;
  /**
   * Tooltip animation
   */
  animation?: TooltipAnimation;
  /**
   * Animation duration
   */
  duration?: number | [number, number];
  /**
   * Display custom HTML as tooltip content
   */
  tooltipHTML?: JSX.Element;
  /**
   * Theme color
   */
  theme?: TooltipTheme;
}

const TooltipComponent = (
  {
    theme = 'material',
    children,
    placement = 'top',
    text,
    arrow,
    animation = 'scale',
    duration = [500, 0],
    tooltipHTML,
  }: TooltipProps,
  ref: TooltipProps['ref'],
) => {
  return (
    <Tippy
      theme={theme}
      animation={animation}
      duration={duration}
      arrow={arrow}
      placement={placement}
      ref={ref}
      disabled={!text}
      content={tooltipHTML ? tooltipHTML : <Text text={text} as="p" textSize="small" />}
    >
      {children}
    </Tippy>
  );
};

export const Tooltip = forwardRef<HTMLDivElement, TooltipProps>(TooltipComponent);
