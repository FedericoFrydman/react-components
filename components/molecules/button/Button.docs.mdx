import { ArgsTable, Story, Canvas } from '@storybook/addon-docs';
import { Button } from './Button';

# Button

Buttons are used to initiate an action. It contains a combination of label, and can be supplemented with an icon.
Button label should be concise and clearly describe action occurs after user’s interaction.

<ArgsTable of={Button} />

## Button Variants

There are three Button variants: `primary`, `secondary`, and `tertiary`. They indicate the heirachary of
actions types and relative importance in the UI. There are three former variants that have been deprecated
but are supported for now: `standard`, `link`, and `destructive`. These will be removed in a future version.

<p>&nbsp;</p>

### Primary Button

---

**Primary button** calls attention to important actions. Use them to emphasize a preferred action. Not all interfaces require a primary button.

<Canvas>
  <Story id="controls-button--primary" />
</Canvas>

- **Do** Limit use of Primary Button on a UI to build visual hierarchy.
- **Do** use Primary Button for submit a form or go to next step in the process.
- **Do not** use Primary Button to open external link.

<p>&nbsp;</p>

### Secondary Button

---

**Secondary button** are secondary buttons for lesser consequential actions, such as Edit or Cancel. Often these are placed alongside a Primary button.

A screen can contain multiple secondary buttons.

<Canvas>
  <Story id="controls-button--secondary" />
</Canvas>

- **Do not** use Secondary Button to open external link

<p>&nbsp;</p>

### Tertiary button

---

**Tertiary Button** is used to navigate to another page and should open in the current window unless:

- Information may be lost, e.g. when someone is filling out a form
- The destination is an external site, e.g. for a knowledge base article

<Canvas>
  <Story id="controls-button--tertiary" />
</Canvas>

- **Do** use Tertiary button for navigational action, such as “Back” or “Next”

<p>&nbsp;</p>

## Destructive Buttons

A Destructive button indicates in the UI that the associated action will delete or remove a relevant item.
When the `destructive` prop is `true`, the button will turn red to indiciate that action type. This can be used with any of the button variants.

### Primary

<Canvas>
  <Story id="controls-button--primary-destructive" />
</Canvas>

### Secondary

<Canvas>
  <Story id="controls-button--secondary-destructive" />
</Canvas>

### Tertiary

<Canvas>
  <Story id="controls-button--tertiary-destructive" />
</Canvas>

## Icon Buttons

<p>&nbsp;</p>

### Icon-only Button

---

**Icon only Button** is a standalone button without any visible text label. Extra care is needed when choosing an icon making sure it is well understood and follow common convention (example: magnify glass for search, person icon for user, trash for delete). This button type is often used to trigger an overflow menu on mobile.

Alt text and other accessibility considerations need to be included to make sure this button makes sense to all users.

<Canvas>
  <Story id="controls-button--icon-button" />
</Canvas>

- **Do** include alt text so user can hover and understand
- **Do** follow common well-known convention when choosing an icon

<p>&nbsp;</p>

### Button with Icon

---

Buttons can have one or two icons, which can present on the left and/or right side of the button.

#### Left Side

<Canvas>
  <Story id="controls-button--btn-w-icon-left" />
</Canvas>

#### Right Side

<Canvas>
  <Story id="controls-button--btn-w-icon-right" />
</Canvas>

#### Custom Color

<Canvas>
  <Story id="controls-button--btn-w-colored-icon-left" />
</Canvas>

<p>&nbsp;</p>

### Stretch Button

---

<Canvas>
  <Story id="controls-button--stretch" />
</Canvas>

- **Do** use `stretch` prop on Button to have more visual impact

<p>&nbsp;</p>

### Button with an icon

---

Buttons can have an icon, the icon can be on the left or right
These is how the btn looks with the icon

- To the **left**
  <Canvas>
    <Story id="controls-button--btn-w-icon-left" />
  </Canvas>
  <p>&nbsp;</p>
- To the **right**
  <Canvas>
    <Story id="controls-button--btn-w-icon-right" />
  </Canvas>
- To the **left with color**
  <Canvas>
    <Story id="controls-button--btn-w-colored-icon-left" />
  </Canvas>

<p>&nbsp;</p>

## General Usage

- Use only one primary button per screen
- Always place buttons in easy-to-spot locations
- Avoid modifying or disabling interactive states
- Where possible, make system actions like copying links and opening in new tabs available
- When pairing Primary and Standard buttons together such as “Send” and “Cancel”, our buttons are right aligned and Primary button should be the farthest right in the group.
  <p>&nbsp;</p>

## Label Considerations

- Write the label in concise, clear terms so that users know what will happen if they click.
- Use title case in the button text.
- Aim for a low character count (four words or fewer).
- Check for text legibility and sufficient contrast when using transparent button styles.
