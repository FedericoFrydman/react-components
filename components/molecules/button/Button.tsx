import { forwardRef, Ref } from 'react';
import cn from 'classnames';

import './button.scss';
import { Icon, IconAndBadgeSizes, IconNames } from '../../atoms/icon/Icon';
import { IconColors } from '../../assets/icons';
import { Text, TextSize } from '../../atoms/text/Text';

export type ButtonShape = 'round' | 'rectangular';

export type ButtonType = 'button' | 'submit' | 'reset';

export type ButtonVariant = 'primary' | 'secondary' | 'tertiary' | 'standard' | 'destructive' | 'link';

export interface ButtonProps extends Omit<React.HTMLProps<HTMLButtonElement>, 'type' | 'css'> {
  /**
   * The text to be shown in the button
   */
  label?: string;
  /**
   * Style of the button. Values are `primary`, `secondary`, `tertiary`, `standard`, `destructive`, and `link`.
   * The following have been deprecated: `standard`, `destructive`, `link`. Please use other variants instead and set the destructive prop to true for destructive variants.
   * @default primary
   */
  variant?: ButtonVariant;
  /**
   * Indicates the button has a destructive action and sets the color to red.
   * @default false
   */
  destructive?: boolean;
  /**
   * Visually and functionally disable the button.
   * @default false
   */
  disabled?: boolean;
  /**
   * Shape of the button - either round or rectangular.
   * @default rectangular
   */
  shape?: ButtonShape;
  /**
   * Stretch the button across the full width of its parent.
   * @default false
   */
  stretch?: boolean;
  /**
   * Display an icon in addition to the text to help to identify the action. The color will be same of the text
   */
  iconLeft?: IconNames;
  /**
   *  Display an icon on the right
   */
  iconRight?: IconNames;
  /**
   * Icon size
   * @default small
   */
  iconSize?: IconAndBadgeSizes;
  /**
   * Applies a color to the icon.
   */
  iconColor?: IconColors;
  /**
   * Text size of the button
   * @default large
   */
  textSize?: TextSize;
  /**
   * The HTML button type
   * @default button
   */
  type?: ButtonType;
  /**
   * Custom class for component.
   */
  classNames?: cn.Argument;
  /**
   * Define the button width
   */
  widthLong?: string;
  /**
   * The ref to the HTML DOM element.
   */
  ref?: Ref<HTMLButtonElement>;
  /**
   * An event id that will be used by analytics and test automation.
   */
  pplsiEventId?: string;
}

const ButtonComponent = (
  {
    variant = 'primary',
    destructive = false,
    disabled = false,
    stretch = false,
    shape = 'rectangular',
    type = 'button',
    iconLeft,
    iconRight,
    iconColor,
    iconSize = 'small',
    textSize = 'large',
    label,
    onClick,
    classNames = [],
    widthLong,
    pplsiEventId,
    ...props
  }: ButtonProps,
  ref: ButtonProps['ref'],
) => {
  // Backwards compatibility for secondary buttons
  if (variant === 'standard') {
    variant = 'secondary';
    console.warn('standard variant has been deprecated. Please use `secondary` instead.');
  }

  // Backwards compatibility for link buttons
  if (variant === 'link') {
    variant = 'tertiary';
    console.warn('link variant has been deprecated. Please use `tertiary` instead.');
  }

  // Backwards compatibility for destructive buttons
  if (variant === 'destructive') {
    destructive = true;
    variant = 'primary';
    console.warn('destructive variant has been deprecated. Please use `destructive` instead.');
  }

  const classButton = 'lsux-button';
  const cnButton = cn(
    {
      [`${classButton}`]: true,
      [`${classButton}--${variant}`]: variant,
      [`${classButton}--${shape}`]: shape,
      [`${classButton}--destructive`]: destructive,
      [`${classButton}--stretch`]: stretch,
      [`${classButton}--icon-only`]: ((!iconLeft && iconRight) || (iconLeft && !iconRight)) && !label,
    },
    classNames,
  );

  const variantColor = variant === 'primary' || destructive ? 'N00' : 'default';
  const color = iconColor ? iconColor : variantColor;
  const dynamicWidth = { width: widthLong };
  const setStyle = widthLong ? dynamicWidth : {};

  const cnText = iconLeft || iconRight ? [cn({ 'pl-3': iconLeft, 'pr-3': iconRight })] : [];

  return (
    <button
      {...props}
      type={type}
      style={{ ...setStyle, ...props.style }}
      data-pplsi-event-id={pplsiEventId}
      className={cnButton}
      disabled={disabled}
      ref={ref}
      onClick={onClick}
    >
      {iconLeft && <Icon name={iconLeft} color={color} size={iconSize} classNames={['lsux-button--icon-left']} />}
      {label && <Text textSize={textSize} textWeight="semibold" text={label} classNames={cnText} />}
      {iconRight && <Icon name={iconRight} color={color} size={iconSize} classNames={['lsux-button--icon-right']} />}
    </button>
  );
};

/**
 * The Button component enables the user to perform an action or navigate
 * to a different screen.
 */

export const Button = forwardRef<HTMLButtonElement, ButtonProps>(ButtonComponent);
