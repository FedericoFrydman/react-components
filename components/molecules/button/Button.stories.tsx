import { Story, Meta } from '@storybook/react';

import { Button, ButtonProps } from './Button';
import docs from './Button.docs.mdx';
import { Container } from '../../../components/atoms/container/Container';
import { ICON_NAMES } from '../../../components/assets/icons';

const ButtonStory: Story<ButtonProps> = (args: ButtonProps) => (
  <Container classNames={['p-3']}>
    <Button {...args} />
  </Container>
);

export const Primary = ButtonStory.bind({});
Primary.args = {
  destructive: false,
  label: 'Primary Button',
  onClick: () => console.log('primary test'),
  pplsiEventId: '123',
  variant: 'primary',
};

export const Secondary = ButtonStory.bind({});
Secondary.args = {
  destructive: false,
  label: 'Secondary Button',
  onClick: () => console.log('secondary test'),
  variant: 'secondary',
};

export const Tertiary = ButtonStory.bind({});
Tertiary.args = {
  label: 'Tertiary Button',
  onClick: () => console.log('tertiary test'),
  variant: 'tertiary',
};

export const PrimaryDestructive = ButtonStory.bind({});
PrimaryDestructive.args = {
  destructive: true,
  label: 'Primary Destructive',
  variant: 'primary',
};

export const SecondaryDestructive = ButtonStory.bind({});
SecondaryDestructive.args = {
  destructive: true,
  label: 'Secondary Destructive',
  variant: 'secondary',
};

export const TertiaryDestructive = ButtonStory.bind({});
TertiaryDestructive.args = {
  destructive: true,
  label: 'Tertiary Destructive',
  variant: 'tertiary',
};

export const IconButton = ButtonStory.bind({});
IconButton.args = {
  iconColor: 'P200',
  iconLeft: 'action_add',
  variant: 'secondary',
};

export const BtnWIconLeft = ButtonStory.bind({});
BtnWIconLeft.args = {
  iconLeft: 'action_add',
  label: 'Primary',
  variant: 'primary',
};

export const BtnWIconRight = ButtonStory.bind({});
BtnWIconRight.args = {
  iconRight: 'action_add',
  label: 'Primary',
  variant: 'primary',
};

export const BtnWColoredIconLeft = ButtonStory.bind({});
BtnWColoredIconLeft.args = {
  iconColor: 'R300',
  iconLeft: 'action_add',
  label: 'Colored icon',
  variant: 'secondary',
};

export const Stretch = ButtonStory.bind({});
Stretch.args = {
  label: 'Stretch Button',
  stretch: true,
  variant: 'secondary',
};

export default {
  argTypes: {
    destructive: { control: 'boolean' },
    disabled: { control: 'boolean' },
    iconLeft: { control: { type: 'select' }, options: ICON_NAMES },
    iconRight: { control: { type: 'select' }, options: ICON_NAMES },
    iconSize: {
      control: { type: 'select' },
      options: ['xsmall', 'small', 'medium-small', 'medium', 'large', 'xlarge'],
    },
    label: { control: { type: 'text' } },
    shape: { control: { type: 'select' }, options: ['rectangular', 'round'] },
    stretch: { control: { type: 'boolean' } },
    textSize: {
      control: { type: 'select' },
      options: ['tiny', 'small', 'medium', 'large', 'extra-large', 'body', 'description'],
    },
    variant: { control: { type: 'select' }, options: ['primary', 'secondary', 'tertiary', 'link'] },
  },
  component: ButtonStory,
  parameters: {
    docs: { page: docs },
  },
  title: 'Controls/Button',
} as Meta;
