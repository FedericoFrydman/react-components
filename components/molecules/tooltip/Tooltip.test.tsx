import { shallow, ShallowWrapper } from 'enzyme';
import { Tooltip } from './Tooltip';
import { Button } from '../../molecules/button/Button';

describe('Tooltip Component', () => {
  let wrapper: ShallowWrapper;

  beforeEach(() => {
    wrapper = shallow(
      <Tooltip theme="light" animation="scale" duration={500} arrow placement="top" text="Some text">
        <Button label="Hover" />
      </Tooltip>,
    );
  });

  test('should render the component', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
