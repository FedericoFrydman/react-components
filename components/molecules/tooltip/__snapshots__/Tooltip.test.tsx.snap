// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltip Component should render the component 1`] = `
<ForwardRef(TippyWrapper)
  animation="scale"
  arrow={true}
  content={
    <ForwardRef(TextComponent)
      as="p"
      text="Some text"
      textSize="small"
    />
  }
  disabled={false}
  duration={500}
  placement="top"
  theme="light"
>
  <ForwardRef(ButtonComponent)
    label="Hover"
  />
</ForwardRef(TippyWrapper)>
`;
